### PR TITLE
added extension-points to targets "compile", "jar", "release" and "nbm"

### PR DESCRIPTION
**Pull request by nigjo**
Since Version "???" Ant has a feature called "extension point". The NetBeans build system can profit from this feature. Currently only one "legacy extension point" is available "netbeans-extra". This PR adds a "-pre"- and "-post" extension-points to the targets "compile", "jar", "release" and "nbm".

A blog post about my thoughts (in german):
https://blog.nigjo.de/netbeans/blog/2018/06/01/extension-points-in-ant/

### CODE REVIEW

**Comment by geertjanw** [comment]
Very cool. I think @jglick should take a look at this.

**Comment by geertjanw** [comment]
Can you take a look at this @jtulach?

**Comment by geertjanw** [comment]
When the items listed above by @jtulach are done, we can merge this.

File: harness/apisupport.harness/release/README [hunk 1]
```diff
@@ -1,1363 +1,1363 @@
-INTRODUCTION
-------------
```
Comment (JaroslavTulach) [inline harness/apisupport.harness/release/README]: The diff shouldn't remove everything and add it again. Try to fix your line endings.

**Comment by geertjanw** [comment]
@nigjo, is this ready and can be merged?

**Comment by matthiasblaesing** [comment]
@nigjo are you still interested in this? I checked the changes and they look sane (I did a double rebase and then I get a single minimal diff)

**Comment by nigjo** [comment]
I whould like to see this in master. We are using this change in our RCP for a while now without any issues. (other than we can now use ant extension points)

**Comment by matthiasblaesing** [comment]
@nigjo ok, changeset was rebased and squashed and is now applied to master: https://github.com/apache/netbeans/commit/f22b10b0b640033d9ba5e54180cb276d91f447cb . Thank you for your contribution.

**Pull request closed without merging**